### PR TITLE
Stop raising Cairo VM errors for Stack over/under-flow

### LINKED ISSUE #776: feat: stop reverting the Cairo VM in the Stack
### Describe the Feature Request

Revert the context but not the Cairo VM

### Describe Preferred Solution

Make under and over flows in opcodes and revert the ctx instead.

### If the feature request is approved, would you be willing to submit a PR? Yes

/ No _(Help can be provided if you need assistance submitting a PR)_

### PR DESCRIPTION
**Pull request by ClementWalter**
Time spent on this PR: 1 day

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

In stack.cairo, one checks for each push/pop op that there is enough and not too many items in the Stack, and possibly raises a Cairo VM errors.

Resolves #776

## What is the new behavior?

No more Cairo error for Stack over/under-flow, but a reverted ExecutionContext.
No extra checks: checks are made only once in the beginning of the opcode.
Stack has been updated to use a `Dict<Uint256*>`, dividing by two the number of dict ops.

### CODE REVIEW

**Comment by ClementWalter** [comment]
Regarding the newly failing ef-tests, we have

previously passing due to some Cairo VM error:
    stRandom2::test_randomStatetest481_d0g0v0_Shanghai
    stRandom2::test_randomStatetest538_d0g0v0_Shanghai
    stRandom2::test_randomStatetest597_d0g0v0_Shanghai
    stRandom2::test_randomStatetest649_d0g0v0_Shanghai
    stRandom::test_randomStatetest257_d0g0v0_Shanghai
    stRandom::test_randomStatetest303_d0g0v0_Shanghai
    stRandom::test_randomStatetest32_d0g0v0_Shanghai
    stReturnDataTest::test_returndatacopy_after_failing_callcode_d0g0v0_Shanghai

previsouly passing and regression
    stSystemOperationsTest::test_suicideAddress_d0g0v0_Shanghai
    stSystemOperationsTest::test_suicideSendEtherPostDeath_d0g0v0_Shanghai
    stSystemOperationsTest::test_suicideSendEtherToMe_d0g0v0_Shanghai
    stTransactionTest::test_SuicidesAndSendMoneyToItselfEtherDestroyed_d0g0v0_Shanghai
    
I'll investigate the suicide issue, but I think that it's fine to skip the others

File: src/kakarot/instructions/comparison_operations.cairo
Comment (Eikix) [inline src/kakarot/instructions/comparison_operations.cairo:363]: why do we need this

File: src/kakarot/instructions/comparison_operations.cairo
Comment (Eikix) [inline src/kakarot/instructions/comparison_operations.cairo:486]: should we do: 
let popped_items_number = 2;
let stack_underflow = is_le(ctx.stack.size, popped_items_number - 1);
// ...
stack.pop_n(.. n=popped_items_number);?

to avoid errors?

File: tests/src/kakarot/instructions/test_stop_and_arithmetic_operations.cairo
Comment (Eikix) [inline tests/src/kakarot/instructions/test_stop_and_arithmetic_operations.cairo:30]: ahh so the difference here is whether or not we use a "new" keyword, yielding either the pointer or the value?

File: tests/utils/helpers.py
Comment (Eikix) [inline tests/utils/helpers.py:43]: why do we delete this

**Review by Eikix — DISMISSED** [review]
lgtm, waiting for oral approval to approve, 
but lgtm

File: tests/src/kakarot/instructions/test_system_operations.cairo
Comment (ClementWalter) [reply tests/src/kakarot/instructions/test_system_operations.cairo:785]: forgotten from previous refacto

File: tests/utils/helpers.py
Comment (ClementWalter) [reply tests/utils/helpers.py:43]: not needed anymore, see the `test_kakarot.py`, we have `result.stack_values` directly

File: tests/utils/helpers.cairo
Comment (ClementWalter) [reply tests/utils/helpers.cairo:168]: not used anymore

File: src/kakarot/instructions/comparison_operations.cairo
Comment (ClementWalter) [reply src/kakarot/instructions/comparison_operations.cairo:363]: to have a pointer, we need either to define `__fp__` or to create a new one

Comment (ClementWalter) [reply src/kakarot/instructions/comparison_operations.cairo:363]: as discussed, I'm adding now a `push_uint256(item: Uint256)`

**Review by Eikix — APPROVED** [review]

**Pull request merged**
